haproxy-ddos-protection-configurator/charmcraft.yaml

```
@@ -0,0 +1,72 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+type: charm
+base: ubuntu@24.04
+build-base: ubuntu@24.04
+
+platforms:
+  amd64:
+
+parts:
+  charm:
+    source: .
+    plugin: uv
+    build-snaps:
+    - astral-uv
+
+name: haproxy-ddos-protection-configurator
+title: HAProxy DDoS protection configurator
+description: |
+  A [Juju](https://juju.is/) [charm](https://juju.is/docs/olm/charmed-operators)
+  that serves as a configurator for HAProxy to provide DDoS protection capabilities.
+
+  HAProxy is a TCP/HTTP reverse proxy which is particularly suited for high
+  availability environments. Since the HAProxy charm currently has many configuration options,
+  this configurator charm aims to simplify the deployment of HAProxy with DDoS by separating the
+  DDoS configuration from the main HAProxy charm.
+
+summary: DDoS protection configurator for HAProxy charm.
+links:
+  documentation: 
+    https://discourse.charmhub.io/t/haproxy-ddos-protection-configurator-documentation/19426
+  issues: https://github.com/canonical/haproxy-operator/issues
+  source: https://github.com/canonical/haproxy-operator/haproxy-ddos-protection-configurator
+  contact:
+  - https://launchpad.net/~canonical-is-devops
+
+assumes:
+- juju >= 3.3
```

Unless there's a strong reason to go for older versions, I'd try to stick to the latest LTS.
```suggestion
- juju >= 3.6
```